### PR TITLE
TDL-17936: Add missing tap-tester tests

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-17936: Add missing tap-tester tests
- Added Tap Tester tests
  - Discovery test
  - Pagination test
  - Automatic fields test
  - All fields test
  - Bookmarks test
  - Start date test

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    def get_type(self):
+        return "platform.tiktok"
+
+    def get_credentials(self):
+        return {
+            "access_token": os.getenv("TAP_TIKTOK_ADS_ACCESS_TOKEN"),
+            "accounts": eval(os.getenv("TAP_TIKTOK_ADS_ACCOUNTS"))
+        }
+
+    def get_properties(self, original: bool = True):
```
Comment (dbshah1212) [inline tests/base.py:42]: Do add function comments.

File: tap_tiktok_ads/__init__.py [hunk 2]
```diff
@@ -16,7 +16,8 @@ def main():
     args = utils.parse_args(REQUIRED_CONFIG_KEYS)
 
     with TikTokClient(access_token=args.config['access_token'],
-                      user_agent=args.config['user_agent']) as tik_tok_client:
+                      user_agent=args.config['user_agent'],
+                      sandbox=args.config.get('sandbox', False)) as tik_tok_client:
```
Comment (dbshah1212) [inline tap_tiktok_ads/__init__.py:20]: Why do we need to add this change in tap-tester PR? If it is necessary then do mention PR dependency.

File: tap_tiktok_ads/streams.py [hunk 3]
```diff
@@ -191,6 +191,7 @@ def __init__(self,
         self.state = state
         self.config = config
         self.client = client
+        self.page_size = int(config.get('page_size', 1000))
```
Comment (dbshah1212) [inline tap_tiktok_ads/streams.py:194]: Any reason to configure the Page_size?

File: tests/test_bookmarks.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import copy
+import os
+from dateutil.parser import parse
+from base import TiktokBase
+from tap_tester import runner, connections, menagerie
+
+def get_second_start_date(stream):
```
Comment (dbshah1212) [inline tests/test_bookmarks.py:7]: Do add function comments

File: tap_tiktok_ads/streams.py (hunk 3 above)
Comment (harshpatel4crest) [reply tap_tiktok_ads/streams.py:194]: Yes, page size from the Tap is passed as `1000`, but we cannot create more than 999 records for the `campaign` stream. POST call is giving the error like: `"The number of campaigns has reached the maximum limit. Please remove a campaign and try again"`. Also for sandbox, the records for **ad_insights** stream the records are automatically generated. Thus we have added a page size param.

File: tap_tiktok_ads/__init__.py (hunk 2 above)
Comment (harshpatel4crest) [reply tap_tiktok_ads/__init__.py:20]: We have added this config param as we have a sandbox account and we need to pass `"sandbox": True`.

File: tests/test_bookmarks.py (hunk 4 above)
Comment (harshpatel4crest) [reply tests/test_bookmarks.py:7]: Added function comments.

File: tests/base.py (hunk 1 above)
Comment (harshpatel4crest) [reply tests/base.py:43]: Added function comments.

File: tap_tiktok_ads/__init__.py (hunk 2 above)
Comment (dbshah1212) [reply tap_tiktok_ads/__init__.py:20]: For that we also have seprate card as well, right?

Comment (harshpatel4crest) [reply tap_tiktok_ads/__init__.py:20]: Yes, it's TDL-18053.

**Review by dbshah1212 — APPROVED** [review]

**Comment by dbshah1212** [comment]
Approved but merge this after the PR: https://github.com/singer-io/tap-tiktok-ads/pull/7 else there will be no meaning of PR: https://github.com/singer-io/tap-tiktok-ads/pull/7

**Pull request merged**